### PR TITLE
c7n-mailer | Slack Dependency Fixes

### PR DESCRIPTION
**Pull request by LykinsN**
I've reworked some of the Slack delivery code so the SlackClient package has been removed altogether, and it now leverages the requests library internal to Boto. Also tweaked some logging statements that were throwing exceptions.

### CODE REVIEW

File: tools/c7n_mailer/c7n_mailer/slack_delivery.py [hunk 1]
```diff
@@ -102,31 +100,34 @@ def retrieve_user_im(self, email_addresses):
                     list[address] = self.caching.get(address)
                     continue
 
-            response = self.client.api_call(
-                "users.lookupByEmail", email=address)
+            response = requests.post(
```
Comment (thisisshi) [inline tools/c7n_mailer/c7n_mailer/slack_delivery.py]: better to do `requests.post(...).json()` here as to lower the number of changes

File: tools/c7n_mailer/c7n_mailer/slack_delivery.py [hunk 2]
```diff
@@ -14,20 +14,18 @@
 import json
 import time
 
-import requests
+from botocore.vendored import requests
```
Comment (kapilt) [inline tools/c7n_mailer/c7n_mailer/slack_delivery.py]: fwiw, the datadog module still depends on requests, so its unclear this really removes a dep, unless we try except our import of it. our lambda packaging subsys is a little baroque for adding package data files, but nutshell certifi pkg's ca pem file needs to be specified by path to make it into the zip.

Comment (kapilt) [reply tools/c7n_mailer/c7n_mailer/slack_delivery.py]: definitely removing slack client removes a few other dependencies (websocket pkg as well), just wanted to note that requests isn't per se one of them.

**Comment by kapilt** [comment]
heh, why'd you close this one?, it still seems worthwhile to remove the slackclient dep.

**Comment by LykinsN** [comment]
I wasn't sure if this was still needed, considering the dependency refactor corrected the lambda failure. I can reopen this one if there's still value from it. 👍

**Comment by LykinsN** [comment]
Rolling these changes into https://github.com/capitalone/cloud-custodian/pull/2347 and closing.

**Pull request closed without merging**